lib/Backend/amd64/LinearScanMD.cpp

```diff
@@ -318,12 +318,20 @@ LinearScanMD::GenerateBailOut(IR::Instr * instr, __in_ecount(registerSaveSymsCou
         this->linearScan->RecordUse(stackSym->scratch.linearScan.lifetime, firstInstr, nullptr, true);
     }
 
+
+
+
     // Load the bailout target into rax
     //     mov  rax, BailOut
     //     call rax
-    Assert(instr->GetSrc1()->IsHelperCallOpnd());
-    Lowerer::InsertMove(IR::RegOpnd::New(nullptr, RegRAX, TyMachPtr, func), instr->GetSrc1(), instr);
-    instr->ReplaceSrc1(IR::RegOpnd::New(nullptr, RegRAX, TyMachPtr, func));
+
+    // TODO: This only applies for current BailOutHelperCall
```

Comment:
>/ TODO: This only applies for current BailOutHelperCal [](start = 5, length = 54)

is this still a todo?

Reply:
you're already not calling this method for lazy bailouts

---
In reply to: [241620360](https://github.com/Microsoft/ChakraCore/pull/5861#discussion_r241620360) [](ancestors = 241620360)

Reply:
I think that's there to remind me to verify if that is the right check to do (probably is). Previously we do those unconditionally. I'll update the comment to reflect it.